addons/account_voucher/migrations/9.0.1.0/post-migration.py

```diff
@@ -71,9 +71,21 @@ def create_payments_from_vouchers(env):
         SET payment_id = av.id
         FROM account_voucher_line avl
         JOIN account_voucher av ON av.id = avl.voucher_id
-        WHERE avl.move_line_id=aml.id
+        WHERE avl.move_line_id = aml.id
         AND av.voucher_type IN ('receipt', 'payment')
-        AND av.state in ('draft', 'posted')
+        AND av.state IN ('draft', 'posted')
+        AND avl.reconcile
```

Comment:
In fact the correct query that should replace this one is:
`        WITH Q1 AS
        (SELECT av.id as av_id, aml.id as aml_id
        FROM account_move_line aml
        INNER JOIN account_move am
        ON am.id = aml.move_id
        INNER JOIN account_voucher av
        ON av.move_id = am.id)
        UPDATE account_move_line aml
        SET payment_id = av.id
        FROM account_voucher_line avl
        JOIN account_voucher av ON av.id = avl.voucher_id
        WHERE avl.move_line_id = aml.id
        AND av.voucher_type IN ('receipt', 'payment')
        AND av.state IN ('draft', 'posted')
        AND (aml.id IN (SELECT credit_move_id FROM account_partial_reconcile
        WHERE debit_move_id IN (SELECT aml_id FROM Q1 WHERE av_id = av.id))
        OR aml.id IN (SELECT debit_move_id FROM account_partial_reconcile
        WHERE credit_move_id IN (SELECT aml_id FROM Q1 WHERE av_id = av.id)))`